lib/types/Authorization.types.ts

```diff
@@ -33,14 +38,27 @@ export interface AccessTokenRequestOpts {
   issuanceInitiation: IssuanceInitiationWithBaseUrl;
   asOpts?: AuthorizationServerOpts;
   metadata?: EndpointMetadata;
+  codeVerifier?: string; // only required for authorization flow
+  code?: string; // only required for authorization flow
+  redirectUri?: string; // only required for authorization flow
   pin?: string; // Pin-number. Only used when required
 }
 
 export interface AuthorizationRequest {
   response_type: ResponseType.AUTH_CODE;
   client_id: string;
+  code_challenge: string;
+  code_challenge_method: CodeChallengeMethod;
   redirect_uri: string;
-  scope?: string;
+  scope?: string[];
+}
+
+export interface AuthorizationRequestOpts {
+  clientId: string;
+  codeChallenge: string;
+  codeChallengeMethod: CodeChallengeMethod;
+  redirectUri: string;
+  scope?: string[];
```

Comment:
A scope is a string which is whitespace separated in case of multiple values
https://www.rfc-editor.org/rfc/rfc6749.html#appendix-A.4

Since these are arguments, I would expect this to at least accept a string, with maybe an optional string[], the end result however should be that it gets converted to a whitespace separated string

Also from the same spec:
````
 The value of the scope parameter is expressed as a list of space-
   delimited, case-sensitive strings.  The strings are defined by the
   authorization server.  If the value contains multiple space-delimited
   strings, their order does not matter, and each string adds an
   additional access range to the requested scope.
````

Happy for us to make these changes btw

Reply:
Done! See other comments for a description.